pac4j-oauth/src/main/java/org/pac4j/oauth/client/FacebookClient.java

```diff
@@ -81,6 +83,8 @@
     protected boolean requiresExtendedToken = false;
     
     protected StateApi20 api20;
+
+    protected String appsecretProof = null;
```

Comment:
Reading that: https://www.sammyk.me/the-single-most-important-way-to-make-your-facebook-app-more-secure, I understand the appsec_proof is another value than the app secret.

Taking a closer look at your code, I see that you added this parameter on the exchange token url (it's not linked to the profile retrieval), which is good. But you should also add it to the profile url parameter https://github.com/narrative-technologies/pac4j/blob/develop/pac4j-oauth/src/main/java/org/pac4j/oauth/client/FacebookClient.java#L128

I guess you need some private method here to add the appsec_proof if it is not null to an url.